src/features/vote/RevealPhase.tsx

```diff
@@ -0,0 +1,262 @@
+                    Curabitur sed eleifend erat. Duis in ante nisi. Donec ut
+                    nibh id justo faucibus fermentum id id ex. Mauris
+                    sollicitudin consequat neque.
+                  </div>
+                </td>
+                <td>
+                  <div>
+                    <p className="vote">{el.vote}</p>
+                  </div>
+                </td>
+                <td>
+                  <div>
+                    {el.revealed
+                      ? "Revealed"
+                      : el.vote !== UNDEFINED_VOTE && !el.revealed
+                      ? "Reveal"
+                      : el.vote === UNDEFINED_VOTE
+                      ? "Uncommitted"
+                      : null}
+                  </div>
+                </td>
+              </tr>
+            );
+          })}
+        </tbody>
+      </table>
+      <div className="end-row">
+        <div className="end-row-item">
+          {round.snapshotId === "0" ? (
+            <Button
+              onClick={() => {
+                if (!signer || !votingContract || !provider) return;
+                votingContract.functions["snapshotMessageHash"]().then(
+                  (hash) => {
+                    const sigHash = hash[0];
+                    if ((window as any).ethereum) {
+                      const mm = (window as any).ethereum;
+                      const Web3 = new web3(mm);
+
+                      // Make sure we use the hot address if the are using a two key contract.
+                      let va = votingAddress;
+                      if (hotAddress) va = hotAddress;
+                      if (va) {
+                        getMessageSignatureMetamask(Web3, sigHash, va).then(
+                          (msg) => {
+                            snapshotCurrentRound(votingContract, msg).then(
+                              (tx) => {
+                                // TODO: Refetch state after snapshot.
+                                console.log("success?", tx);
+                              }
+                            );
+                          }
+                        );
+                      }
+                    }
+                  }
+                );
+              }}
+              variant="secondary"
+            >
+              {signer ? "Snapshot Round" : "Connect Wallet to Snapshot"}
+            </Button>
+          ) : postRevealData.length ? (
+            <Button
+              type="button"
+              onClick={() => {
+                console.log("PSR", postRevealData);
+                // Make sure to use the two key contract for revealing if it exists
+                let vc = votingContract;
+                if (designatedVotingContract) vc = designatedVotingContract;
+                if (vc) {
+                  revealVotes(vc, postRevealData).then((res) => {
+                    // refetch votes.
+                    console.log("successfully revealed");
```

Comment:
i dont wnat to forget to remove these logs, line 232 and 214.  We might want a TODO on having a way to notify user of success